### PR TITLE
[java] Add missing support events for Web Driver Listener 

### PR DESCRIPTION
**Pull request by dev-velo**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

adds explicit target locator events to the listener support classes. This can already be done implicitly since events can resolve to the web driver instance but this exposes these events explicitly
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: java/src/org/openqa/selenium/support/events/WebDriverListener.java [hunk 1]
```diff
@@ -312,4 +313,48 @@ default void afterMaximize(WebDriver.Window window) {}
...
+
+  /**
+   * @param targetLocator the target locator being used for the action
+   * @param nameOrHandle The name of the window or the handle as returned by {@link
+   *     org.openqa.selenium.WebDriver#getWindowHandle()} or <code>null</code> if switching to a new
+   *     window created by {@link org.openqa.selenium.WebDriver.TargetLocator#newWindow(WindowType)}
+   * @param driver WebDriver
+   */
+  default void afterWindow(
+      WebDriver.TargetLocator targetLocator, String nameOrHandle, WebDriver driver) {}
```
Comment (valfirst) [inline java/src/org/openqa/selenium/support/events/WebDriverListener.java:1105]: would you mind adding hooks for all methods from `TargetLocator`?

**Comment by titusfortner** [comment]
I think we want to add for everything in this: https://github.com/seleniumhq/selenium/blob/trunk/java/src/org/openqa/selenium/WebDriver.java#L451

**Comment by titusfortner** [comment]
@valfirst & @RevealOscar thanks for your work and interest on this. Just sending a reminder that we want to get this into the next release if either of you have bandwidth to work on it this week. Thanks again.

**Comment by martingrossmann** [comment]
For my understanding:
Is `afterWindow` the new version of `WebDriverEventListener.afterSwitchToWindow()`?

**Comment by titusfortner** [comment]
Yes it pulls the name of the actual method being called, which is window()

**Comment by titusfortner** [comment]
@valfirst & @RevealOscar another ping on getting this added before 4.17 release, which will be "soon"
Thanks again for your help!

**Comment by valfirst** [comment]
@titusfortner I don't have permissions to work on this PR.

if @RevealOscar doesn't have capacity to complete the PR, I would merge it as is (it's better than nothing). Then next contributor can add missing functionalities in a separate PR.

**Comment by diemol** [comment]
@RevealOscar could you please resolve the conflicts and then I can merge this?

**Comment by titusfortner** [comment]
it was just imports conflicting, if there's a problem RBE will fail 😂

**Review by valfirst — APPROVED** [review]

**Review by diemol — APPROVED** [review]
Thank you all!

**Pull request merged**